packages/shared/src/forms/SliderLabels.js

```diff
@@ -20,30 +19,38 @@ type Props = {|
   +min: number,
 |};
 
+const LABEL_MARGIN = parseInt(defaultTokens.spaceXSmall, 10);
+
 export default function SliderLabels(props: Props) {
   const [width, setWidth] = React.useState(0);
   const [labelStartWidth, setLabelStartWidth] = React.useState(0);
-  const [labelStartAtMax, setLabelStartAtMax] = React.useState(false);
   const [labelEndWidth, setLabelEndWidth] = React.useState(0);
   const [paddingLeft, setPaddingLeft] = React.useState(0);
   const [paddingRight, setPaddingRight] = React.useState(0);
 
-  useCalculateSliderPositions({
-    labelEndWidth,
-    labelStartAtMax,
-    labelStartWidth,
-    paddingLeft,
-    paddingRight,
-    props,
-    width,
-    setPaddingLeft,
-    setLabelStartAtMax,
-    setPaddingRight,
-    min: props.min,
-    max: props.max,
-    endValue: props.endValue,
-    startValue: props.startValue,
-  });
+  React.useEffect(() => {
+    const maxOffsetLeft =
+      width - labelStartWidth - labelEndWidth - LABEL_MARGIN + paddingRight;
+    const offsetLeft = Math.max(
+      (props.startValue / props.max) * width - labelStartWidth / 2,
+      0,
+    );
+    setPaddingLeft(Math.min(maxOffsetLeft, offsetLeft));
+  }, [props.startValue, labelStartWidth]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  React.useEffect(() => {
+    if (props.endValue != null) {
+      const maxRightOffset =
```

Comment:
```suggestion
      const maxOffsetRight =
```
to be more consistent in the naming